programs/protocol-contracts-solana/src/lib.rs

```diff
@@ -367,73 +364,30 @@ pub mod gateway {
             Errors::SPLAtaAndMintAddressMismatch,
         );
 
-        // test whether the recipient_ata is created or not; if not, create it
+        // test whether the recipient_ata is created or not; if not log and exit successfully
         let recipient_ata_account = ctx.accounts.recipient_ata.to_account_info();
         if recipient_ata_account.lamports() == 0
             || *recipient_ata_account.owner == ctx.accounts.system_program.key()
         {
             // if lamports of recipient_ata_account is 0 or its owner being system program then it's not created
-            msg!(
-                "Creating associated token account {:?} for recipient {:?}...",
-                recipient_ata_account.key(),
-                ctx.accounts.recipient.key(),
-            );
-            let signer_info = &ctx.accounts.signer.to_account_info();
-            let bal_before = signer_info.lamports();
-            invoke(
-                &create_associated_token_account(
-                    ctx.accounts.signer.to_account_info().key,
-                    ctx.accounts.recipient.to_account_info().key,
-                    ctx.accounts.mint_account.to_account_info().key,
-                    ctx.accounts.token_program.key,
-                ),
-                &[
-                    ctx.accounts.mint_account.to_account_info().clone(),
-                    ctx.accounts.recipient_ata.clone(),
-                    ctx.accounts.recipient.to_account_info().clone(),
-                    ctx.accounts.signer.to_account_info().clone(),
-                    ctx.accounts.system_program.to_account_info().clone(),
-                    ctx.accounts.token_program.to_account_info().clone(),
-                    ctx.accounts
-                        .associated_token_program
-                        .to_account_info()
-                        .clone(),
-                ],
-            )?;
-            let bal_after = signer_info.lamports();
-
-            msg!("Associated token account for recipient created!");
-            msg!(
-                "Refunding the rent paid by the signer {:?}",
-                ctx.accounts.signer.to_account_info().key
+            msg!("recipient ATA account does not exist");
```

Comment:
thinking about how zetaclient should parse and validate this--
grep the log? 
can it be tricked? (for example, the user included two intructions in the same tx, one is from their attack program **which emits the same log above**, and the second instruction is the withdraw spl that actually points to a valid ata address so transfer actually succeeds)